opendota.py

```diff
@@ -8,8 +8,133 @@
+                else: 
+                    select = select + f"player_matches{i}.hero_id AS hero{i+1}, \n"
+                i = i+1
+            i = 0
+            for hero in data: 
+                if i == 0: 
+                    select = select + f"player_matches.player_slot AS ps1, \n"
+                else: 
+                    select = select + f"player_matches{i}.player_slot AS ps{i+1}, \n"
+                i = i+1
+
+        select = select + f"matches.match_id,\n"
+        select = select + f"leagues.name leaguename\n"
+        return select
+
+    def join(*heroes): 
+        """generates join statements for certain num of heroes"""
+        join = f"FROM player_matches \n"
+        i = 0
+        for data in heroes: 
+            for hero in data:
+                if i == 0: 
+                    i = i+1
+                    continue
+                else:
+                    join = join + f"JOIN player_matches AS player_matches{i}\n"
+                    join = join + f"ON player_matches.match_id = player_matches{i}.match_id\n"
+                    i = i+1
+        join = join + f"JOIN matches\n"
+        join = join + f"ON player_matches.match_id = matches.match_id\n"
+        join = join + f"JOIN leagues using(leagueid)\n"
+        return join
+
+    def where(*heroes): 
+        """generates where statement for certain num of heroes"""
+        i = 0
+        where = f"WHERE"
+        for data in heroes: 
+            for hero in data: 
+                if i == 0: 
+                    where = where + f" player_matches.hero_id = {get_hero_id(hero)}\n"
+                    i = i+1
+                else: 
+                    where = where + f"AND player_matches{i}.hero_id = {get_hero_id(hero)}\n"
+                    i = i+1
+
+        #probably need to pass in a timestamp or something at some point for the following
+        where = where + f"AND matches.start_time >= extract(epoch FROM timestamp '2018-10-01T06:53:44.537Z')\n"
+        return where
+
+    def teams(*heroes): 
+        """adds onto where statement and specifies which team each hero is on, hero1+hero2 =teamA, hero3+hero4 = teamB"""
+        i = 0
+        teams = ""
+        for data in heroes: 
+            for hero in data: 
+                if i == 0: 
+                    i = i+1
+                    continue
+                elif i == 1: 
+                    teams = teams + f"AND abs(player_matches.player_slot - player_matches1.player_slot) < 6\n"
+                    i = i+1
+                elif i == 2: 
+                    teams = teams + f"AND abs(player_matches.player_slot - player_matches2.player_slot) > 123\n"
+                    i = i+1
+                elif i == 3: 
+                    teams = teams + f"AND abs(player_matches2.player_slot - player_matches3.player_slot) < 6\n"
+                    teams = teams + f"AND abs(player_matches1.player_slot - player_matches3.player_slot) > 123\n"
+                    i = i+1
```

Comment:
i think we say if it's the first hero in team A, no teams clause. if the hero is >1 item in team A, add same team clause. if in team B, add other team clause, if that makes sense.